File: app/src/main/java/com/example/tdm_android/managers/UserManager.java
package com.example.tdm_android.managers;


import android.content.Context;

import androidx.annotation.NonNull;

import com.example.tdm_android.helpers.DBHelper;
import com.example.tdm_android.models.User;
import com.j256.ormlite.android.apptools.OpenHelperManager;
import com.j256.ormlite.android.apptools.OrmLiteSqliteOpenHelper;
import com.j256.ormlite.dao.Dao;

import java.sql.SQLException;
import java.util.List;

public class UserManager {

    private static UserManager instance;
    Dao<User, Integer> dao;

    public static UserManager getInstance(@NonNull Context context) {
        if(instance == null) {
            instance = new UserManager(context);
        }
        return instance;
    }

    public UserManager(Context context) {
        OrmLiteSqliteOpenHelper helper = OpenHelperManager.getHelper(context, DBHelper.class);
        try {
            dao = helper.getDao(User.class);
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public List<User> getUsers() throws Exception {
        return dao.queryForAll();
    }

    public void createUser(User user) throws Exception {
        dao.create(user);
    }

    public User getOneUserByField(String field, String username) throws Exception {
        return dao.queryForEq(field, username).get(0);
    }

}

File: app/src/main/java/com/example/tdm_android/managers/UserManager.kt
package com.example.tdm_android.managers

import android.content.Context
import com.j256.ormlite.android.apptools.OrmLiteSqliteOpenHelper
import com.j256.ormlite.dao.Dao
import kotlin.Throws
import com.j256.ormlite.android.apptools.OpenHelperManager
import com.example.tdm_android.helpers.DBHelper
import com.example.tdm_android.models.User
import java.lang.Exception
import java.sql.SQLException

class UserManager(context: Context?) {
    private var dao: Dao<User?, Int>? = null

    @get:Throws(Exception::class)
    val users: List<User?>
        get() = dao!!.queryForAll()

    @Throws(Exception::class)
    fun createUser(user: User?) {
        dao!!.create(user)
    }

    @Throws(Exception::class)
    fun getOneUserByField(field: String?, username: String?): User? {
        return dao!!.queryForEq(field, username)[0]
    }

    @Throws(Exception::class)
    fun updateUser(user: User?) {
        dao!!.update(user)
    }

    companion object {
        private var instance: UserManager? = null
        fun getInstance(context: Context): UserManager? {
            if (instance == null) {
                instance = UserManager(context)
            }
            return instance
        }
    }

    init {
        val helper: OrmLiteSqliteOpenHelper =
            OpenHelperManager.getHelper(context, DBHelper::class.java)
        try {
            dao = helper.getDao(
                User::class.java
            )
        } catch (e: SQLException) {
            e.printStackTrace()
        }
    }
}